targets.py

```diff
@@ -2,34 +2,44 @@
 
 
 class DirectoryTarget:
-    def __init__(self, path):
-        self.name = path
+    def __init__(self, name, path):
+        self.name = name
+        self.path = path
 
     @property
     def id(self):
         return "Directory: " + self.name
 
+    @property
+    def mount_id(self):
+        return self.path
+
     def before_restore(self):
         pass
 
 
 class NamedVolumeTarget:
-    def __init__(self, name):
+    def __init__(self, name, volume):
         self.name = name
+        self.volume = volume
 
     @property
     def id(self):
         return "Named volume: " + self.name
 
+    @property
+    def mount_id(self):
+        return self.volume
+
     def _volume_exists(self):
         text = run(
             ["docker", "volume", "ls", "-q"],
```

Comment:
can you use the docker python client instead?

```python
self.volume in [x.name for x in self.docker.volumes.list()]
```

and in the object initialisation either pass in a docker client or construct with `docker.client.from_env()`)

Reply:
(or just use `self.docker.volumes.get(self.volume)` with a `except docker.errors.NotFound:` clause)

Reply:
Opened https://vimc.myjetbrains.com/youtrack/issue/VIMC-1590 for this